### PR TITLE
gh-76002: asyncio: allow pause_reading() from connection_made()

### PR DESCRIPTION
**Pull request by pitrou**
* Issue: gh-76002

### CODE REVIEW

**Comment by 1st1** [comment]
Hm.  We'll need to run tests from the uvloop (has more functional tests) project with this patch.

**Comment by 1st1** [comment]
But I like the change.  Let me think about it for a few days.

**Comment by pitrou** [comment]
No pb!

**Comment by pitrou** [comment]
@1st1 any update here?

File: Lib/asyncio/selector_events.py [hunk 1]
```diff
@@ -683,14 +683,16 @@ def __init__(self, loop, sock, protocol, waiter=None,
         # decreases the latency (in some cases significantly.)
         _set_nodelay(self._sock)
 
-        self._loop.call_soon(self._protocol.connection_made, self)
-        # only start reading when connection_made() has been called
-        self._loop.call_soon(self._loop._add_reader,
-                             self._sock_fd, self._read_ready)
-        if waiter is not None:
+        def _call_connection_made():
```
Comment (asvetlov) [inline Lib/asyncio/selector_events.py:686]: Convert nested function into instance method maybe?

**Comment by asvetlov** [comment]
I like the PR but the came approach should be added to all transports (SSL and pipes) IMHO.

**Comment by vstinner** [comment]
I removed the " needs backport to 3.6" label, the 3.6 branch no longer accept bugfixes (only security fixes are accepted): https://devguide.python.org/#status-of-python-branches

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]
There are merge conflicts.

**Comment by kumaraditya303** [comment]
Closing as there are merge conflicts.

**Pull request closed without merging**